tunix/rl/agentic/TrajectoryCollector/trajectory_collect_engine.py

```diff
@@ -0,0 +1,135 @@
+# tunix/rl/multi_turn/execution/trajectory_collector.py
+import asyncio
+import time
+from typing import Any, Callable, Dict, Optional, Tuple, List, AsyncGenerator
+
+from tunix.rl.agentic.agents.base_agent import BaseAgent, Trajectory
+from tunix.rl.agentic.environments.base_environment import BaseEnv
+
+
+class TrajectoryCollectEngine:
+    """
+    A ready-to-use rollout sampler:
+      • Implements complete rollout logic (reset → step* → final reward → return)
+      • Allows custom model inference and final reward via callbacks
+      • Supports concurrent multi-pair rollout (streaming results like rllm)
+    """
+
+    # --------------------------------------------------
+    # Constructor
+    # --------------------------------------------------
+    def __init__(
+        self,
+        agent: BaseAgent,
+        env: BaseEnv,
+        *,
+        model_call: Callable[[list[Dict[str, str]]], str],
+        final_reward_fn: Optional[Callable[[Dict, str], float]] = None,
+        max_steps: int = 10,
+        gamma: float = 1.0,
+        timeout: float = 30.0,
```

Comment:
add pydocs please, for example, timeout for what, for a single turn rollout or model call or environment interaction or sth else? and what's the unit of timeout. And what is gemma for, etc